src/operator/nn/mkldnn/mkldnn_batch_norm-inl.h

```diff
@@ -332,17 +333,18 @@ void MKLDNNBatchNormBackward(const nnvm::NodeAttrs &attrs, const OpContext &ctx,
     const NDArray &beta     = in_data[batchnorm::kBeta];
     DType *weight_buf = reinterpret_cast<DType *>(bwd.GetWeight().get_data_handle());
     nnvm::dim_t channels_ = data.shape()[1];
-    for (int i = 0; i < channels_; i++) {
-      if (!param.fix_gamma)
-        weight_buf[i] = (gamma.data().dptr<DType>())[i];   // weight
-      else
+    DType *weight_ptr = reinterpret_cast<DType *>(gamma.data().dptr<DType>());
+    DType* bias_ptr = reinterpret_cast<DType *>(beta.data().dptr<DType>());
```

Comment:
It seems `reinterpret_cast<DType*>` is not needed.

Reply:
exactly. Done